docxcompose/properties.py

```diff
@@ -225,12 +225,69 @@ def update(self, name, value):
         cfield = xpath(
             self.doc.element.body,
             u'.//w:instrText[contains(.,\'DOCPROPERTY "{}"\')]'.format(name))
+
         if cfield:
             w_p = cfield[0].getparent().getparent()
+
+            def get_runs():
```

Comment:
as just discussed: a good option would be to first find all elements that belong to a field together, then update them in a second step. that should hopefully reduce the xpath complexity.

Reply:
This was all refactored, so let's say this is 👍